django/contrib/admin/sites.py

```diff
@@ -106,7 +106,12 @@ def register(self, model_or_iterable, admin_class=None, **options):
                 )
 
             if model in self._registry:
-                raise AlreadyRegistered('The model %s is already registered' % model.__name__)
+                raise AlreadyRegistered(
+                    'Cannot register model %s because is already registered with %s.' % (
+                        model.__name__,
+                        self._registry[model].__class__.__name__,
```

Comment:
I'd say omitting `__class__.__name__` is more useful as it'll include the `app_label`. I would just omit `'.ModelAdmin'` if the string ends with that, since that's probably `register()` without a `ModelAdmin` subclass.

So in case 1, the error would look like: The model %s is already registered with app 'admin_registration'.
In case 2: The model %s is already registered with 'admin_registration.PersonAdmin'.